packages/compiler/test/server/completion.test.ts

```diff
@@ -1370,6 +1370,268 @@ describe("identifiers", () => {
+        subValue: int16;
+      }
+      model TestModel {
+        name: string;
+        value: TestSubModel;
+      }
+      model MyLogContext<T extends TestModel, P extends valueof TestModel> {
+      }
+      model MyLogContext2<
+        T extends {name: string, value: {key: string, arr: TestModel[]}},
+        P extends valueof {name: string, value: {key: string, arr: TestModel[]}}>{
+      }
+      model MyLogContext3<T extends TestModel[], P extends valueof {name: string, value: boolean}[]>{}
+      model MyLogContext4<T, P extends string>{}
+      `;
+
+    it("show all properties for referenced model template parameter", async () => {
+      (
+        await Promise.all(
+          [`alias test = MyLogContext<{┆}`].map(async (item) => await complete(`${def}\n${item}`))
+        )
+      ).forEach((completions) => {
+        check(
+          completions,
+          [
+            {
+              label: "name",
+              insertText: "name",
+              kind: CompletionItemKind.Field,
+              documentation: {
+                kind: "markdown",
+                value: "(model property)\n```typespec\nTestModel.name: string\n```",
+              },
+            },
+            {
+              label: "value",
+              insertText: "value",
+              kind: CompletionItemKind.Field,
+              documentation: {
+                kind: "markdown",
+                value: "(model property)\n```typespec\nTestModel.value: TestSubModel\n```",
+              },
+            },
+          ],
+          {
+            fullDocs: true,
+            allowAdditionalCompletions: false,
+          }
+        );
+      });
+    });
+
+    it("show all properties for referenced literal object template parameter", async () => {
+      (
+        await Promise.all(
+          [`alias test = MyLogContext<{}, #{value: #{┆}}`].map(
+            async (item) => await complete(`${def}\n${item}`)
+          )
+        )
+      ).forEach((completions) => {
+        check(
+          completions,
+          [
+            {
+              label: "subName",
+              insertText: "subName",
+              kind: CompletionItemKind.Field,
+              documentation: {
+                kind: "markdown",
+                value: "(model property)\n```typespec\nTestSubModel.subName: string\n```",
+              },
+            },
+            {
+              label: "subValue",
+              insertText: "subValue",
+              kind: CompletionItemKind.Field,
+              documentation: {
+                kind: "markdown",
+                value: "(model property)\n```typespec\nTestSubModel.subValue: int16\n```",
+              },
+            },
+          ],
+          {
+            fullDocs: true,
+            allowAdditionalCompletions: false,
+          }
+        );
+      });
+    });
+
+    it("show all properties for referenced model expression parameter", async () => {
+      (
+        await Promise.all(
```

Comment:
I don't know if that's what you did previously but if things go wrong this makes it very hard to debug. Tests are already run in parallel

Reply:
updated to use it.each